src/samples/http/download.c

```diff
@@ -65,6 +66,8 @@ response_func(void *userdata, const struct sol_http_client_connection *connectio
     fd = NULL;
     pending = NULL;
 
+    SOL_HTTP_RESPONSE_CHECK_API(response);
+
     if (response->response_code != SOL_HTTP_STATUS_OK) {
```

Comment:
This is a callback that gives you the type, no need to check it. 

Same for others in this file